File: src/main/java/io/protobase/f7/formulas/math/RAND.java
package io.protobase.f7.formulas.math;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;

public class RAND extends AbstractFormula {
  public static FormulaName NAME = FormulaName.RAND;
  public static RAND SELF = new RAND();

  @Override
  public Object internal(GridColumnRowKey origin, Object... values) {
    checkLength(values.length, 0, NAME);
    return Math.random();
  }
}

File: src/main/js/formulas/math/RAND.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class RAND extends AbstractFormula {
  static SELF: RAND = new RAND();
  NAME = FormulaName.RAND;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 0, this.NAME);
    return Math.random();
  }
}
